### PR TITLE
core: create TRACE_INSECURE log level

### LINKED ISSUE #8778: CephObjectStore debugHTTPClient leaks admin credentials
The `debugHTTPClient` in the Ceph object store code can leak access credentials. See an operator log snippet below:

```
2021-09-21 17:41:27.430246 D | ceph-object-controller: creating s3 user object "rook-ceph-internal-s3-
user-checker-73dc21e6-95ec-49d6-a392-1ddeea6739de" for object store "ocs-storagecluster-cephobjectstor
e" health check
2021-09-21 17:41:27.430464 D | ceph-object-controller: 
GET /admin/user?display-name=rook-ceph-internal-s3-user-checker-73dc21e6-95ec-49d6-a392-1ddeea6739de&f
ormat=json&uid=rook-ceph-internal-s3-user-checker-73dc21e6-95ec-49d6-a392-1ddeea6739de HTTP/1.1
Host: rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc
User-Agent: Go-http-client/1.1
Authorization: AWS4-HMAC-SHA256 Credential=6ACH4N455IY3234IIFVP/20210921/default/s3/aws4_request, Sign
edHeaders=host;x-amz-content-sha256;x-amz-date, Signature=406e147687b0cc172c1fb6f49125a46ca5a4385af31f
9668499461550b4fea85
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20210921T174127Z
Accept-Encoding: gzip


2021-09-21 17:41:27.475161 D | ceph-object-controller: 
HTTP/1.1 200 OK
Content-Length: 834
Connection: Keep-Alive
Date: Tue, 21 Sep 2021 17:41:27 GMT
X-Amz-Request-Id: tx000000000000000000004-00614a1947-1bce1-ocs-storagecluster-cephobjectstore

{"tenant":"","user_id":"rook-ceph-internal-s3-user-checker-73dc21e6-95ec-49d6-a392-1ddeea6739de","disp
lay_name":"rook-ceph-internal-s3-user-checker-73dc21e6-95ec-49d6-a392-1ddeea6739de","email":"","suspen
ded":0,"max_buckets":1000,"subusers":[],"keys":[{"user":"rook-ceph-internal-s3-user-checker-73dc21e6-9
5ec-49d6-a392-1ddeea6739de","access_key":"FAS2UVXRT8WTQDXOK89J","secret_key":"i9oZ97wZODWuzNfBNCTPixI3
MVbV831j5tqBQbXg"}],"swift_keys":[],"caps":[],"op_mask":"read, write, delete","system":"false","admin"
:"false","default_placement":"","default_storage_class":"","placement_tags":[],"bucket_quota":{"enable
d":false,"check_on_raw":false,"max_size":-1,"max_size_kb":0,"max_objects":-1},"user_quota":{"enabled":
false,"check_on_raw":false,"max_size":-1,"max_size_kb":0,"max_objects":-1},"temp_url_keys":[],"type":"
rgw","mfa_ids":[]}
```

### Initial analysis 
1. `"keys"` in the body at the bottom definitely leaks credentials. 

2. It is unclear to me whether the header log also is leaking sensitive credential info, but I do see that the `AWS4-HMAC-SHA256 Credential=6ACH4N455IY3234IIFVP/20210921/default/s3/aws4_request` line is consistent throughout the operator logs. This suggests to me that we should consider it sensitive info.

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

**Expected behavior:**

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

### PR DESCRIPTION
**Pull request by BlaineEXE**
Create a new log level for Rook that is hidden from users. This is the
most verbose log level, and it is the level developers would like to use
to get debug logs that are important for debugging but that could leak
senstivie information like credentials in production use.

If a user sets their debug level to "TRACE", they will merely get
"DEBUG" level logs. Only if they set "TRACE_INSECURE" will they get
trace logs, and those are likely to include insecure information. Rook
tries very hard not to leak sensitive information in logs even with
verbose "DEBUG" logs.

Resolves #8778

Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: cmd/rook/rook/rook.go
Comment (BlaineEXE) [inline cmd/rook/rook/rook.go:65]: Was totally unused.

File: pkg/util/logging.go
Comment (BlaineEXE) [inline pkg/util/logging.go:46]: Create this to allow reusing the logic.

File: pkg/daemon/ceph/client/config_test.go
Comment (BlaineEXE) [inline pkg/daemon/ceph/client/config_test.go:64]: This file was the only usage of `LogLevel` from `clusterd.Context`, so I removed it.

File: cluster/examples/kubernetes/ceph/operator-openshift.yaml
Comment (travisn) [inline cluster/examples/kubernetes/ceph/operator-openshift.yaml:104]: I'm not aware of Rook using `critical` or `notice` anywhere. Shall we remove those options from the docs if there is no real application?

File: pkg/util/logging.go
Comment (travisn) [inline pkg/util/logging.go:39]: nit to avoid var names too short
```suggestion
	logLevel, err := capnslog.ParseLevel(userLogLevelSelection)
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: cluster/examples/kubernetes/ceph/operator-openshift.yaml
Comment (BlaineEXE) [reply cluster/examples/kubernetes/ceph/operator-openshift.yaml:104]: That sounds reasonable.

**Review by travisn — COMMENTED** [review]
LGTM, pending rebase to pick up the CI update

**Comment by BlaineEXE** [comment]
@Mergifyio rebase

File: pkg/util/logging.go
Comment (leseb) [inline pkg/util/logging.go:25]: Can we have unit tests for this? Otherwise LGTM

**Review by leseb — CHANGES_REQUESTED** [review]

**Review by leseb — APPROVED** [review]

**Comment by BlaineEXE** [comment]
@Mergifyio rebase

**Review by travisn — APPROVED** [review]

**Pull request merged**
